airflow/upgrade/rules/chain_between_dag_and_operator_not_allowed_rule.py

```diff
@@ -36,7 +36,7 @@ def _change_info(self, file_path, line_number):
 
     def _check_file(self, file_path):
         problems = []
-        with open(file_path, "r") as file_pointer:
+        with open(file_path, "r", errors="ignore") as file_pointer:
```

Comment:
Should we check file extension before opening it and open only `.py` files?

Reply:
I think this may solve the problem. To be honest I'm not sure if ignoring errors is a good idea - it may make hard spotting a next issue.

Reply:
Yeah. Fixed. Please take a look https://github.com/apache/airflow/pull/14344/commits/03b2581fcb4f27b2000e6c4490246d07148304e9

Reply:
Do you need also a "Unable to read python file" message like added in https://github.com/apache/airflow/pull/13371 ?

Reply:
Done